### PR TITLE
Extract missing gcs_to_local example DAG from gcs example

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This PR closes one of the issues in #8280
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
@ephraimbuddy Can you add system tests also? This will ensure that this example is valid.

**Comment by ephraimbuddy** [comment]
> @ephraimbuddy Can you add system tests also? This will ensure that this example is valid.

I added it but it sees it as invalid. I'll add it again and fix the problem

**Comment by ephraimbuddy** [comment]
@mik-laj, Can you check if you can run the system test, I'm having error related to docker at the moment. On PID.

```
INFO     tests.utils.logging_command_executor.LoggingCommandExecutor:logging_command_executor.py:41 Stderr: docker: --pid: invalid PID mode.
See 'docker run --help'.
docker: --pid: invalid PID mode.
See 'docker run --help'.
```

**Comment by mik-laj** [comment]
@ephraimbuddy I fixed this tests. 

docker-in-docker causes various problems. I personally install gcloud locally in such a situation.
You can do it in three steps.
```
rm $(command -v gcloud)
curl http://sdk.cloud.google.com | CLOUDSDK_CORE_DISABLE_PROMPTS=1 bash -
source /root/google-cloud-sdk/path.bash.inc
```

**Comment by ephraimbuddy** [comment]
Wow. Thank you!

File: tests/providers/google/cloud/transfers/test_gcs_to_local_system.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+@pytest.mark.credential_file(GCP_GCS_KEY)
+class GoogleCloudStorageExampleDagsTest(GoogleSystemTest):
+    @provide_gcp_context(GCP_GCS_KEY)
+    def setUp(self):
+        super().setUp()
+        self.create_gcs_bucket(BUCKET)
+        self.upload_content_to_gcs(
+            lines=f"{os.urandom(1 * 1024 * 1024)}", bucket=BUCKET, filename=PATH_TO_REMOTE_FILE
+        )
+        self.execute_cmd(["gsutil", "ls", f"gs://{BUCKET}"])
```
Comment (ephraimbuddy) [inline tests/providers/google/cloud/transfers/test_gcs_to_local_system.py]:
```suggestion
        
```

**Review by mik-laj — APPROVED** [review]

**Pull request merged**